kustomization/go.mod

```diff
@@ -0,0 +1,66 @@
+module github.com/fluxcd/pkg/kustomization
+
+go 1.17
+
+require (
+	github.com/drone/envsubst v1.0.3
```

Comment:
I think we should switch to `github.com/drone/envsubst/v2` see https://github.com/drone/envsubst/pull/27

Note that `v1.0.3` is not what we use in kustomize-controller https://github.com/fluxcd/kustomize-controller/blob/main/go.mod#L11

Reply:
Done. 

This might break some existing Users yaml files that are compensating for the escaping bug fixed in `github.com/drone/envsubst/v2` . So we have to keep it in mind when updating the controller dependency.